src/core/event_registry/server_event.rs

```diff
@@ -150,6 +159,30 @@ impl ServerEventAppExt for App {
 
         self
     }
+
+    fn make_independent<E: Event>(&mut self) -> &mut Self {
+        self.world_mut()
+            .resource_scope(|world, mut event_registry: Mut<EventRegistry>| {
+                let events_id = world
+                    .components()
+                    .resource_id::<Events<E>>()
+                    .unwrap_or_else(|| {
+                        panic!(
+                            "event `{}` should be previously registered",
+                            any::type_name::<E>()
+                        )
+                    });
+                let marked = event_registry.make_independent(events_id);
```

Comment:
I this we should just do nothing if it's already marked. To make it consistent with `register_type()` from Bevy.

Reply:
Fixed